src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -21,16 +21,27 @@
     UniqueTagList getTags();
 
     /**
-     * Returns true if the values inside this object is same as those of the other
+     * Returns true if this person is the same as that represented in another
+     * object. Two people are the same if their identity fields (name and
+     * telephone) are the same.
+     */
+    default boolean isSamePerson(ReadOnlyPerson other) {
+        return other == this || (other != null && other.getName().equals(this.getName())
+                && other.getPhone().equals(this.getPhone()));
+    }
+
+    /**
+     * Returns true if all data in this object is the same as that in another
      * (Note: interfaces cannot override .equals)
      */
-    default boolean isSameStateAs(ReadOnlyPerson other) {
+    default boolean hasSameData(ReadOnlyPerson other) {
         return other == this // short circuit if same object
                 || (other != null // this is first to avoid NPE below
                 && other.getName().equals(this.getName()) // state checks here onwards
                 && other.getPhone().equals(this.getPhone())
                 && other.getEmail().equals(this.getEmail())
-                && other.getAddress().equals(this.getAddress()));
+                && other.getAddress().equals(this.getAddress())
+                && other.getTags().equals(this.getTags()));
```

Comment:
The `&&` shouldn't be aligned together with `||`, since `&&` are in a deeper condition level than `||`. 

Thus `&&` should be indented one level more.